src/nodes/cannonball.rs

```diff
@@ -0,0 +1,204 @@
+        let cannonball_sprite = AnimatedSprite::new(
+            Self::CANNONBALL_WIDTH as u32,
+            Self::CANNONBALL_HEIGHT as u32,
+            &[Animation {
+                name: Self::CANNONBALL_ANIMATION_ROLLING.to_string(),
+                row: 0,
+                frames: 1,
+                fps: 1,
+            }],
+            true,
+        );
+
+        let speed = if facing {
+            vec2(
+                Self::CANNONBALL_INITIAL_SPEED_X_REL,
+                Self::CANNONBALL_INITIAL_SPEED_Y,
+            )
+        } else {
+            vec2(
+                -Self::CANNONBALL_INITIAL_SPEED_X_REL,
+                Self::CANNONBALL_INITIAL_SPEED_Y,
+            )
+        };
+
+        let mut body = PhysicsBody::new(
+            &mut resources.collision_world,
+            pos,
+            0.0,
+            vec2(Self::CANNONBALL_WIDTH, Self::CANNONBALL_HEIGHT),
+        );
+        body.speed = speed;
+
+        /*PhysicsBody {
+            pos,
+            facing,
+            angle: 0.0,
+            speed,
+            collider: None,
+            on_ground: false,
+            last_frame_on_ground: false,
+            have_gravity: true,
+            bouncyness: 1.0,
+        };*/
+
+        /*let cannonball_mount_pos = if facing {
+            vec2(Self::CANNONBALL_MOUNT_X_REL, Self::CANNONBALL_MOUNT_Y)
+        } else {
+            vec2(-Self::CANNONBALL_MOUNT_X_REL, Self::CANNONBALL_MOUNT_Y)
+        };*/
+
+        /*body.collider = Some(resources.collision_world.add_actor(
+            body.pos + cannonball_mount_pos,
+            CANNONBALL_WIDTH as i32,
+            CANNONBALL_HEIGHT as i32,
+        ));*/
+
+        Self {
+            cannonball_sprite,
+            body,
+            lived: 0.0,
+            countdown: Self::CANNONBALL_COUNTDOWN_DURATION,
+            owner_id,
+            owner_safe_countdown: Self::CANNONBALL_OWNER_SAFE_TIME,
+            //erupting: false,
+            //erupting_enable_on_y: None,
+        }
+    }
+
+    pub fn spawn(pos: Vec2, facing: bool, owner_id: u8) {
+        let cannonball = Cannonball::new(pos, facing, owner_id);
+        scene::add_node(cannonball);
+    }
+}
+
+/*impl EruptedItem for Cannonball {
+    fn spawn_for_volcano(pos: Vec2, speed: Vec2, enable_at_y: f32, owner_id: u8) {
+        let mut cannonball = Self::new(pos, true, owner_id);
+
+        cannonball.lived -= 2.; // give extra life, since they're random
+        cannonball.body.speed = speed;
+        cannonball.body.collider = None;
+        cannonball.erupting = true;
+        cannonball.erupting_enable_on_y = Some(enable_at_y);
+
+        scene::add_node(cannonball);
+    }
+
+    fn body(&mut self) -> &mut PhysicsBody {
+        &mut self.body
+    }
+    fn enable_at_y(&self) -> f32 {
+        self.erupting_enable_on_y.unwrap()
+    }
+}*/
+
+impl scene::Node for Cannonball {
+    fn fixed_update(mut cannonball: RefMut<Self>) {
+        /*if cannonball.erupting {
```

Comment:
Same, commented code :)

Reply:
It's just there for when the volcano is reimplemented, but that will probably need some new code anyways. Removing